File: leetcode-tasks/0002-add-two-numbers.js
"use strict";

/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
  let a = "",
    b = "",
    c = "",
    result = null;

  while (l1) {
    a = l1.val + a;
    l1 = l1.next;
  }
  while (l2) {
    b = l2.val + b;
    l2 = l2.next;
  }

  c = (BigInt(a) + BigInt(b)).toString();

  for (let i = 0; i < c.length; i++) {
    const buff = result;
    result = new ListNode(c[i]);
    result.next = buff;
  }

  return result;
};

File: tasks/0002-add-two-numbers.go
// Accepted 1569 / 1569 testcases passed Sergey Pomortsev submitted at Jul 15, 2026 22:09
// Runtime 0 ms Beats 100.00% Memory 6.73 MB Beats 5.37%

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
  var l1Str, l2Str string

  for link := l1; link != nil; link = link.Next {
    l1Str = strconv.Itoa(link.Val) + l1Str
  }

  for link := l2; link != nil; link = link.Next {
    l2Str = strconv.Itoa(link.Val) + l2Str
  }

  bigL1Int, _ := new(big.Int).SetString(l1Str, 10)
  bigL2Int, _ := new(big.Int).SetString(l2Str, 10)
  l3Str := new(big.Int).Add(bigL1Int, bigL2Int).String()
  list := &ListNode{Val: 0, Next: nil}
  link := list

  for i := len(l3Str) - 1; i >= 0; i-- {
    v, _ := strconv.Atoi(string(l3Str[i]))
    
    var next *ListNode = nil
    if (i != 0) {
      next = &ListNode{Val: 0, Next: nil}
    }
    
    link.Val = v
    link.Next = next
    link = link.Next
  }

  return list
}